<img src="./figures/Logo_KIT.svg" width="250" height="250"  style="zoom:15%;float:right;" />

# Fakultät für Physik 

## Physikalisches Praktikum P2 für Studierende der Physik



Versuch P2-47, 48, 49 (Stand: April 2023)

[Raum F1-08](http://www-ekp.physik.uni-karlsruhe.de/~simonis/praktikum/layoutobjekte/Lageplan_P1.png)

<br>
Name: _________Schwarz_________ Vorname: _______Felix___________ E-Mail: _______unuhz@student.kit.edu___________
<br>
<br>
Name: _________Steib_________ Vorname: ________Lukas__________ E-Mail: ________uzmpb@student.kit.edu__________
<br>
<br>
<br>
Gruppennummer: __Mo-21___
<br>
<br>
<br>
Betreuer: _________David Wößmann_________
<br>
<br>
<br>
Versuch durchgeführt am: ________08.05.2023__________

---

**Beanstandungen:**

<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
Testiert am: __________________ Vermerk: __________________

# Ideales und reales Gas

## Motivation

Die klassische Thermodynamik beschäftigt sich mit den Eigenschaften makroskopisch ausgedehnter Systeme, die durch Zustandsgrößen, wie Volumen ($V$), Druck ($p$) oder Temperatur ($T$) charakterisiert und mit Hilfe von Zustandsgleichungen beschrieben werden. Zustandsänderungen idealer Gase werden durch die [ideale Gasgleichung](https://de.wikipedia.org/wiki/Thermische_Zustandsgleichung_idealer_Gase) oder die [Adiabaten-Gleichungen](https://de.wikipedia.org/wiki/Adiabatische_Zustands%C3%A4nderung) beschrieben; Wärmeänderungen bei konstantem Druck oder Volumen durch die [kalorischen Zustandsgleichungen](https://de.wikipedia.org/wiki/Zustandsgleichung), die die Änderungen der Temperatur ($\mathrm{d}T$) mit der Änderungen der inneren Energie ($\mathrm{d}U$) oder Enthalpie ($\mathrm{d}H$) verknüpfen. Als Potentialgrößen hängen weder $U$ noch $H$ davon ab, wie ein betreffendes System (z.B. im $pV$-Diagramm) in seinen Zustand gelangt ist. Wärme ($Q$) hingegen ist keine Potentialgröße; Sie kann abhängig davon, auf welchem Weg ein System in seinen Zustand (z.B. im $pV$-Diagramm) gelangt zu unterschiedlichen Anteilen entweder zu Änderungen der inneren Energie ($\mathrm{d}U$) oder zur Verrichtung mechanischer Arbeit ($\delta W$) führen, was die Grundlage jeder Wärmekraftmaschine ist. Die Entwicklung der klassischen Thermodynamik als Disziplin der Ingeneurswissenschaften hat ab Mitte des 19. Jahrunderts v.a. durch die Arbeiten von [Nicolas Carnot](https://de.wikipedia.org/wiki/Nicolas_L%C3%A9onard_Sadi_Carnot), [Julius Mayer](https://de.wikipedia.org/wiki/Julius_Robert_Mayer), [James Joule](https://de.wikipedia.org/wiki/James_Prescott_Joule) oder [Rudolf Clausius](https://de.wikipedia.org/wiki/Rudolf_Clausius) rasante Entwicklungen genommen. 

Die hier angebotenen Versuche bieten Ihnen die Möglichkeit sich mit den nicht immer intuitiven Vorgängen der Thermodynamik experimentell vertraut zu machen: Sie führen Temperaturmessungen mit einem Gasthermometer durch, und bestimmen mit verschiedenen Methoden den [Adiabatenexponenten](https://de.wikipedia.org/wiki/Isentropenexponent) $\kappa=c_{p}/c_{V}$, wobei das verwendete Gas jeweils verschiedene Wege im $pV$-Diagramm durchläuft. Schließlich messen Sie eine Dampfdruckkurve aus, bei der sich die Phasenzustände "flüssig" und "gasförmig" im thermodynamischen Gleichgewicht befinden. 

Um diese Experimente erfolgreich durchführen zu können müssen Sie die Zustände des Arbeitsgases achtsam und sorgfältig präparieren. Bei den meisten hierzu verwendeten Geräten handelt es sich um eigens angefertigte Präzisionsinstrumente.

## Lernziele

Wir listen im Folgenden die wichtigsten **Lernziele** auf, die wir Ihnen mit dem Versuch **Ideales und reales Gas** vermitteln möchten: 

- Sie üben sich in der Durchführung thermodynamischer Experimente mit idealen und realen Gasen und präparieren dabei verschiedene Zustandsänderungen des verwendeten Arbeitsgases im $pV$-Diagramm.
- Sie verwenden die ideale Gasgleichung zur Temperaturmessung mit Hilfe eines [Gasthermometers](https://de.wikipedia.org/wiki/Gasthermometer). Dabei ermittlen Sie mit Hilfe des Gasthermometers die Temperatur des absoluten Nullpunkts in $^{\circ}\mathrm{C}$.
- Sie bestimmen $\kappa$ für Luft und ggf. Argon: 
  - einmal nach der Methode von [Clément-Desormes](https://de.wikipedia.org/wiki/Experiment_von_Cl%C3%A9ment-Desormes), unter Herbeiführung *adiabatischer* und *isochorer* Zustandsänderungen; und 
  - einmal nach einer Schwingungsmethode, unter Herbeiführung ausschließlich *adiabatischer* Zustandsänderungen des Arbeitsgases.

- Sie bestimmen die [Dampfdruckkurve](https://de.wikipedia.org/wiki/Dampfdruck) von n-[Hexan](https://de.wikipedia.org/wiki/Hexan) und leiten daraus seine Verdampfungswärme $Q_{\mathrm{M}}$ ab. 

## Versuchsaufbau

Zwei beispielhafte Aufbauten der hier durchzuführenden Versuche finden Sie in den folgenden Abbildungen dargestellt: 

**Gasthermometer:**

<img src="./figures/image014.jpg" width="500" height="500" style="zoom:60%;" />

**Bestimmung von $\kappa$ nach der Methode von Clément-Desormes:**

<img src="./figures/image016.jpg"  width="500" height="500" style="zoom:60%;" />

Für die verschiedenen Versuchsteile stehen Ihnen vier verschiedene Versuchsaufbauten zur Verfügung. Diese enthalten die folgenden wichtigsten Elemente: 

- Ein Gasthermometer mit Becherglas, Heizpilz (Heizleistung $320\,\mathrm{W}$) und Rührvorrichtung.

- Ein [Kathetometer](https://de.wikipedia.org/wiki/Kathetometer) zur exakten Ablesung der Manometer-Anzeigen.

- Zwei Glasflaschen (Volumen, jeweils $V=10\,\mathrm{l}$) für die Aufgaben 2 und 3. 

- Ein U-Rohr-Manometer. 

- Ein Handblasebalg. 

- Ein Präzisionsglasrohr (Länge $l=60\,\mathrm{cm}$) mit exakt eingepasster Stahlkugel (Durchmesser $d=16\,\mathrm{mm}\pm0,5\%$, $m=16,68\,\mathrm{g}\pm0,1\%$).

- Ein [Kolbenprober](https://de.wikipedia.org/wiki/Kathetometer) ($V=100\,\mathrm{ml}$); die Abmessungen des Kolbens sind: Durchmesser $d=31,1(1)\,\mathrm{mm}$, $m=109,80(5)\,\mathrm{g}$ (einschließlich Magnet zur automatischen Frequenzbestimmung).

- Ein Frequenzzählgerät.

- Ein Wassergefäß ($V=2\,\mathrm{l}$). 

- Ein Dampfdruckgefäß, das mit $1\,\mathrm{mol}$ n-Hexan befüllt und direkt mit einem Hg-Manometer verbunden ist.

- Ein Thermometer (Messbereich $0-50^{\circ}\mathrm{C}$).

- Im Versuchsraum ist ferner ein Barometer zur Bestimmung des Umgebungsdrucks vorhanden.

## Wichtige Hinweise

- Bei den hier verwendeten Glasgeräten handelt es sich um empfindliche Spezialanfertigungen und zum Teil um eigens angefertigte Präzisionsmessgeräte. Gehen Sie **vorsichtig mit diesen Geräten** um und halten Sie sie sauber. 
- Beginnen Sie alle im folgenden beschriebenen Versuche **erst nach Rücksprache** mit dem/der Tutor:in. 
- Um Stromschlägen vorzubeugen fassen Sie bei Feuchtigkeit nicht in den eingeschalteten Heizpilz.

## Durchführung

### Aufgabe 1: Gasthermometer

Bestimmen Sie den [Spannungskoeffizienten](https://de.wikipedia.org/wiki/Ausdehnungskoeffizient) $\alpha$ von Luft und berechnen Sie daraus die Temperatur des absoluten Nullpunkts in $^{\circ}\mathrm{C}$. Verwenden Sie hierzu die Zustandsgleichung 

$$
\left.p(T)\right|_{V=const.} = p_{0}\left(1+\alpha\,T\right).
$$

Berechnen Sie den Effekt den die Ausdehnung des Glaskolbens G als Funktion von $T$ hat und korrigieren Sie Ihre Messung entsprechend (siehe [Hinweise für den Versuch "Ideales und reales Gas"](https://git.scc.kit.edu/etp-lehre/p2-for-students/-/blob/main/Ideales_und_reales_Gas/README.md)). 

Um Wartezeiten zu überbrücken können Sie parallel die Messungen zu Aufgabe 2 durchführen.  

---

**Lösung:** \
Für die Bestimmung des Spannungskoeffizienten wird die Höhe der Säulen im Gasthermometer gemessen bei ungefähr $0^{\circ} C$ und bei circa $100^{\circ} C$. \
Damit können wir den Druck berechnen mit der Formel $p(T)=\Delta h \cdot 133.32 + p_{0} $. $p_0$ ist der Atmosphärendruck der bei diesem Versuch immer vorliegt und $\Delta h$ ist die Höhendifferenz zwischen linker und rechter Säule.

In [1]:
import numpy as np #Lade Pakete
import matplotlib.pyplot as plt
from PhyPraKit import readPicoScope, resample
from scipy import signal
from scipy import interpolate
import sys
from kafe2 import  XYContainer, Fit, XYFit, Plot, ContoursProfiler                            
from uncertainties import ufloat, unumpy
import uncertainties.unumpy as unp
from math import log10, floor
import scipy.integrate as integrate
import pandas as pd
import csv

/opt/conda/lib/python3.10/site-packages/kafe2/core/minimizers/__init__.py:34: UserWarning: iminuit is not installed. While not strictly needed its use is heavily recommended for better performance and convergence.
  warnings.warn(
cling::DynamicLibraryManager::loadLibrary(): libGL.so.1: cannot open shared object file: No such file or directory
Error in <AutoloadLibraryMU>: Failed to load library /opt/conda/lib/libEve.so.6.28.00cling JIT session error: Failed to materialize symbols: { (main, { _ZeqRK7TStringRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE }) }


Welcome to JupyROOT 6.28/00


In [2]:
p_0= ufloat(100200,100200*0.01
           )  #pascal 
data=pd.read_csv('Messung/Aufgabe1_kalt.txt',delimiter=';',decimal=".",skiprows=3) #importiere Daten
data.columns=['nkalt','T_kalt','h_linkskalt','h_rechtskalt']
nkalt = data['nkalt'].to_numpy() 
T_kalt = data['T_kalt'].to_numpy()
h_rechtskalt = data['h_rechtskalt'].to_numpy()*10**3
h_linkskalt = data['h_linkskalt'].to_numpy()*10**3
print(data)
#h_rechtskalt = ufloat(h_rechtskalt,h_rechtskalt*0.01)
#h_linkskalt = ufloat(h_linkskalt,h_linkskalt*0.01)
data=pd.read_csv('Messung/Aufgabe1-warm.txt',delimiter=';',decimal=".",skiprows=3) #importiere Daten
data.columns=['nwarm','T_warm','h_linkswarm','h_rechtswarm']
nwarm = data['nwarm'].to_numpy() 
T_warm = data['T_warm'].to_numpy()
h_rechtswarm = data['h_rechtswarm'].to_numpy()*10**3
h_linkswarm = data['h_linkswarm'].to_numpy()*10**3
print(data)
#h_rechtswarm = uflaot
p_kalt = (h_rechtskalt-h_linkskalt)*133.32+p_0
p_kalt_mean = np.sum(p_kalt)/len(p_kalt)
                     
p_warm = (h_rechtswarm-h_linkswarm)*133.32+p_0
p_warm_mean = np.sum(p_warm)/len(p_warm)
print("Beim Gefrierpunkt von Wasser ergibt sich ein Druck von ", p_kalt_mean, "hPascal. Beim Siedepunkt von Wasser ergibt sich ein Druck von ", p_warm_mean, "hPascal.")

   nkalt  T_kalt  h_linkskalt  h_rechtskalt
0      2     1.0        0.090         0.037
1      3     1.0        0.090         0.036
2      4     1.0        0.089         0.036
3      5     1.0        0.089         0.036
   nwarm  T_warm  h_linkswarm  h_rechtswarm
0      2   100.5        0.089         0.299
1      3   100.5        0.089         0.300
2      4   100.5        0.089         0.300
3      5   100.5        0.089         0.300
Beim Gefrierpunkt von Wasser ergibt sich ein Druck von  (9.31+/-0.10)e+04 hPascal. Beim Siedepunkt von Wasser ergibt sich ein Druck von  (1.283+/-0.010)e+05 hPascal.


Mit den gewonnenen Drücken ist der Spannungskoeffizient $\alpha$ mit der Formel $\left.p(T)\right|_{V=const.} = p_{0}\left(1+\alpha\,T\right)$ berechenbar. Wobei $p_0$ der dem Druck bei $0^{\circ} C$ entspricht.
Zuerst wird die Formel noch nach $\alpha$ umgestellt:\
$\Longrightarrow \alpha=\frac{p_{warm}-p_{kalt}}{p_{kalt}\cdot \Delta T} = \frac{p_{warm}-p_{kalt}}{p_{warm}\cdot \left(T_{kalt}-T_{kalt}\right)} $

In [3]:
alpha=(p_warm-p_kalt)/(p_kalt*(T_warm-T_kalt))

Damit ergibt sich für Alpha in erster Näherung ein Wert von $ \frac{1}{^{\circ}C }$.
Beachtet man die thermische Ausdehnung des Glaskolbens erhält man einen besseren Wert.
$\alpha = \alpha^ {(0)} + \frac{p(T_S)}{p_0} \cdot \gamma$

In [4]:
alpha_besser=alpha+p_warm/p_kalt*2.5*10**(-5)

Damit ergibt sich für Alpha ein besserer Wert von $ \frac{1}{^{\circ}C }$.
Über diesen Wert lässt sich der absolute Nullpunkt der Formel $T_{0}=-\frac{1}{\alpha}$ berechnen. 

In [5]:
T_0=-1/alpha_besser
print(np.mean(T_0))

-260.8+/-2.8


Der berechnete Nullpunkt liegt bei $-260.8 \pm 2.8^{\circ}C$. Der Literaturwert für den absoluten Nullpunkt liegt bei $-273,15^{\circ}C$. (https://www.chemie.de/lexikon/Absoluter_Nullpunkt.html) \
Der berechnete Nullpunkt hat eine Abweichung von $4.52\%$ vom Literaturwert.

### Aufgabe 2: Adiabatenexponent (nach [Clément-Desormes](https://de.wikipedia.org/wiki/Experiment_von_Cl%C3%A9ment-Desormes))

Bestimmen Sie $\kappa$ von Luft. Skizzieren Sie den Prozess im $pV$-Diagramm. Führen Sie die Messung mehrmals durch, um ein Maß für die Streuung aufgrund statistischer Fluktuationen zu erhalten. Führen Sie danach zwei weitere Messreihen durch, bei denen Sie das Ventil in Schritt 2 (siehe [Hinweise für den Versuch "Ideales und reales Gas"](https://git.scc.kit.edu/etp-lehre/p2-for-students/-/blob/main/Ideales_und_reales_Gas/README.md)) einmal länger und einmal kürzer geöffnet halten, als dies Ihrer Meinung nach für einen adiabatischen Druckausgleich nötig wäre. Auf diese Weise können Sie den Effekt abschätzen, der dadurch entsteht, dass Sie die Zeitspanne für einen adiabatischen Druckausgleich nicht richtig treffen. Beachten Sie hierzu die Hinweise zu dieser Aufgabe im Dokument [Hinweise für den Versuch "Ideales und reales Gas"](https://git.scc.kit.edu/etp-lehre/p2-for-students/-/blob/main/Ideales_und_reales_Gas/README.md).

---

**Lösung:** \
$\kappa$ lässt sich aus folgender Formel berechnen:\
$\kappa = \frac{h_1}{h_1-h_2}$
Um $\kappa$ möglichst genau zu bestimmen, wird der Mittelwert von mehreren Messreihen gebildet.

In [6]:
h1 = np.array([155-26,160-21,164-16,171-10,166-16])
h2 = np.array([130-51,134-49,145-36,141-40,135-47])
kappa = h1/(h1-h2)
print("kappa =", np.mean(kappa), "+-", np.std(kappa))

kappa = 2.810326808197776 +- 0.4994358912461121


Die Bestimmung von $\kappa$ mit öffnen des Ventils ergibt für Kappa einen Wert von $\kappa=2.8103 \pm 0.4994$. Der Literaturwert für Kappa beträgt 1.40 (https://de.wikipedia.org/wiki/Isentropenexponent). Damit ergibt sich zu dem Literaturwert eine Abweichung von $100\%$.

In [7]:
h1_kurz = np.array([171-10,169-13,163-18])
h2_kurz = np.array([161-20,154-28,141-39])
kappa = h1_kurz/(h1_kurz-h2_kurz)
print("kappa kurz =", np.mean(kappa), "+-", np.std(kappa))

kappa kurz = 5.540697674418605 +- 1.9248826111018942


In [8]:
h1_lang = np.array([167-15,168-15,170-11])
h2_lang = np.array([144-36,141-40,135-46])
kappa = h1_lang/(h1_lang-h2_lang)
print("kappa lang =", np.mean(kappa), "+-", np.std(kappa))

kappa lang = 2.88942723942724 +- 0.48445064867447785


### Aufgabe 3: Adiabatenexponent (Schwingungsmethode)

Bestimmen Sie $\kappa$ indem Sie einen geeigneten Pfropfen über einem Luftpolster in Schwingung versetzen. Die Kompression und Expansion der Luft unter dem Pfropfen erfolgen dabei adiabatisch. Durch dieses Vorgehen lässt sich $\kappa$ aus dem Zusammenhang

$$
T = 2\pi\sqrt{\frac{m\,V}{\kappa\,p\,A^{2}}}
$$

bestimmen, wobei in diesem Fall

- $T$ der Periodendauer der Schwingung(!); 
- $m$ der Masse des Pfropfens; 
- $A$ der Querschnittsfläche des Rohrs entsprechen. 

Nehmen Sie für $V$ das Volumen des untersuchten Luftpolsters unter dem Pfropfen (in dessen Ruhelage) und für $p$ den Umgebungsdruck an.

Für diese Aufgabe können Sie $\kappa$ entweder nach der [Methode von Rüchardt](https://de.wikipedia.org/wiki/R%C3%BCchardt-Experiment), oder mit Hilfe eines Kolbenprobers durchführen. Aufgabenstellung und Auswertung variieren je nach Methode:

- Nach der Methode nach Rüchardt bestimmen Sie $\kappa$ nur für Luft.
- Mit Hilfe des Kolbenprobers bestimmen Sie $\kappa$ für Luft und für Argon. Überprüfen Sie die Ergebnisse mit Ihrer Erwartung aus der [kinetischen Gastheorie](https://de.wikipedia.org/wiki/Kinetische_Gastheorie).  

Beachten Sie hierzu die [Hinweise zum Versuch "Ideales und reales Gas"](https://git.scc.kit.edu/etp-lehre/p2-for-students/-/blob/main/Ideales_und_reales_Gas/README.md).   

---

**Lösung:**



In [9]:
T_gemessen = [1.12,1.16,1.03,1.09,1.16,1.08,1.05,1.1,1.19,0.98,1.19,1.13,
             1.07,1.10,1.08,1.06,1.12,1.07,1.11,1.09,1.14,0.98,1.18,1.07,1.08,1.01,
             1.13,1.02,1.04,1.15,1.12,1.14,1.06,1.06,1.08,1.15,0.98,
              1.09, 1.17, 1.13, 1.11, 1.05, 1.12, 1.04, 1.08, 1.00,
              0.96, 1.17, 1.08, 1.08, 1.09, 1.05, 1.28, 1.02, 1.04
             ] #T IST DIE ZEIT! Fehler auf T beachten?
T = ufloat(np.mean(T_gemessen),np.std(T_gemessen))
m = ufloat(0.01668, 0.01688*0.001)#fehler angegeben
A = ufloat(np.pi * 0.008**2,0.005*np.pi * 0.008**2) 
V = ufloat(10.58 * 10**-3,10.58*0.003*10**-3) #Liter, 0.3% fehler
p = ufloat(100200,100200*0.01
           )  #pascal 
kappa = (2*np.pi/T)**2 * m * V /(p * A**2)
print(kappa)
#kappa aus freiheitsgraden berechnen

1.45+/-0.16


### Aufgabe 4: Dampfdruckkurve (n-Hexan)

Bestimmen Sie die Dampfdruckkurve von $1\,\mathrm{mol}$ n-Hexan zwischen Zimmertemperatur und $0^{\circ}\mathrm{C}$ und bestimmen Sie daraus die molare Verdampfungswärme $Q_{\mathrm{M}}$ von n-Hexan. 

---

**Lösung:**



In [10]:
p_0=ufloat(100200,100200*0.01
           )  #pascal 

In [24]:
h_rechts1 = 0.2817
h_links = np.array([0.3735,0.3642,0.361,0.35715,0.3538,0.3487,0.3455,0.3429,0.3397,0.33625,0.3351,0.3339,0.33495,0.3369,0.3403,0.3413,0.3463,0.3498,0.3525,0.3554,0.3585,0.3625])*10**(3)
T=[24.3,20.0,18.4,16.0,14.2,12.2,10.0,8.2,6.2,4.0,2.1,1.4,2.1,4.1,5.7,9.0,10.2,12.5,14.2,15.9,17.7,20.3] # Temp in Celsius
h_mitte = (h_links[0]+h_rechts1)/2
h_rechts = np.array(2*h_mitte-h_links)
p = (h_rechts-h_links)*133.32+p_0
#print(p)
x=-unp.exp(p/p_0)/8.31446261815324
x_uncer=unp.std_devs(x)
x_true=unp.nominal_values(x)

In [25]:
x_error = x_uncer
#y_error = 0.02
xy_data = XYContainer(x,T)
def lin(T,Q=0.7):
    return T*Q
line_fit=Fit(data=xy_data,model_function=lin)
line_fit.add_error(axis='x', err_val=x_error,relative=False)  # add the x-error to the fit
#line_fit.add_error(axis='y', err_val=y_error,relative=True)
line_fit.do_fit()
line_fit.assign_parameter_names(x='T',a='aT')
line_fit.assign_model_function_expression('Linear')
line_fit.assign_model_function_expression("{a} * {x}")
#line_fit.report()

line_fit.assign_parameter_latex_names(x='T', a='Q')
line_fit.assign_model_function_latex_name('-Exp(p/p0)/R')
line_fit.assign_model_function_latex_expression('{Q} \\cdot {x}')


line_fit.data_container.label = "Verwendete Daten"
line_fit.model_label = "Modellfunktion"

line_fit.data_container.axis_labels = ["-Exp(p/P0)/R in J /(mol K)","T in °C"]

plot = Plot(fit_objects= line_fit)
plot.plot()
plt.show()

TypeError: can't convert an affine function (<class 'uncertainties.core.AffineScalarFunc'>) to float; use x.nominal_value